src/frontend/src/binder/relation/table_or_source.rs

```diff
@@ -161,9 +161,20 @@ impl Binder {
 
     pub(crate) fn bind_table_source(&mut self, name: ObjectName) -> Result<BoundTableSource> {
         let (schema_name, source_name) = Self::resolve_table_name(name)?;
+
         let source = self
             .catalog
-            .get_source_by_name(&self.db_name, &schema_name, &source_name)?;
+            .get_source_by_name(&self.db_name, &schema_name, &source_name)
+            .map_err(|err| {
+                if let Ok(table) = self.catalog.get_table_by_name(&self.db_name, &schema_name, &source_name)
+                && table.associated_source_id().is_none(){
+                    return RwError::from(ErrorCode::InvalidInputSyntax(
+                        format!("cannot change materialized view {}",source_name)
+                    ));
+                }
```

Comment:
I think we should not report error here, `bind_table_source` is not only used in create materiazed view.